lib/util/flattenOperands.js

```diff
@@ -34,293 +35,80 @@ interested in how that works
 // 2+2+2, ie one + node that has three children.
 // Input: an expression tree
 // Output: the expression tree updated with flattened operations
-function flattenOperands(node) {
-  if (Node.Type.isConstant(node, true)) {
-    // the evaluate() changes unary minuses around constant nodes to constant nodes
-    // with negative values.
-    const constNode = Node.Creator.constant(evaluate(node));
-    if (node.changeGroup) {
-      constNode.changeGroup = node.changeGroup;
-    }
-    return constNode;
-  }
-  else if (Node.Type.isOperator(node)) {
-    if ('+-/*'.includes(node.op)) {
-      let parentOp;
-      if (node.op === '/') {
-        // Division is flattened in partner with multiplication. This means
-        // that after collecting the operands, they'll be children args of *
-        parentOp = '*';
-      }
-      else if (node.op === '-') {
-        // Subtraction is flattened in partner with addition, This means that
-        // after collecting the operands, they'll be children args of +
-        parentOp = '+';
+function flattenOperands(root) {
+  return replace(root, {
+    leave(node) {
+      if (query.isAdd(node)) {
+        return build.add(
+          ...node.args.reduce((accum, arg) => {
+            if (query.isAdd(arg)) {
+              return accum.concat(arg.args);
+            }
+            else {
+              return accum.concat(arg);
+            }
+          }, [])
+        );
       }
-      else {
-        parentOp = node.op;
+      else if (query.isMul(node)) {
+        // Rewrite explicit multiplication to be implicit if the factors look
+        // like those of a polynomial.  The first factor must be a number and
```

Comment:
oh! interesting. I'd be open to keeping that, makes it easier to parse sentences from each other